testing/trino-product-tests/src/main/java/io/trino/tests/product/hive/TestCreateDropSchema.java

```diff
@@ -39,18 +41,82 @@
     @Test
     public void testCreateDropSchema()
     {
-        onHive().executeQuery("DROP DATABASE IF EXISTS test_drop_schema CASCADE");
+        String schemaName = "test_drop_schema";
+        String schemaDir = warehouseDirectory + "/test_drop_schema.db";
 
-        onTrino().executeQuery("CREATE SCHEMA test_drop_schema");
-        assertTrue(hdfsClient.exist(warehouseDirectory + "/test_drop_schema.db"));
+        ensureSchemaDoesNotExist(schemaName);
+
+        assertQuerySucceeds("CREATE SCHEMA test_drop_schema");
+        assertThat(hdfsClient.exist(schemaDir))
+                .as("Check if expected schema directory exists after creating schema")
+                .isTrue();
 
         onTrino().executeQuery("CREATE TABLE test_drop_schema.test_drop (col1 int)");
+
         assertQueryFailure(() -> query("DROP SCHEMA test_drop_schema"))
                 .hasMessageContaining("line 1:1: Cannot drop non-empty schema 'test_drop_schema'");
 
         onTrino().executeQuery("DROP TABLE test_drop_schema.test_drop");
 
-        onTrino().executeQuery("DROP SCHEMA test_drop_schema");
-        assertFalse(hdfsClient.exist(warehouseDirectory + "/test_drop_schema.db"));
+        assertQuerySucceeds("DROP SCHEMA test_drop_schema");
+        assertThat(hdfsClient.exist(schemaDir))
+                .as("Check if schema directory exists after dropping schema")
+                .isFalse();
+    }
+
+    @Test
+    public void testDropSchemaDeletesDirectory()
```

Comment:
This depends on file systems.
We should have a copy of that test in `io.trino.plugin.hive.AbstractTestHive`, so it's run for supported storages too.